R/plotBubble.R

```diff
@@ -0,0 +1,45 @@
+#' @title Plot Bubble plot
+#' @description Plot a bubble plot with the color of the plot being the mean expression
+#' and the size of the dot being the percent of cells in the cluster expressing the gene.
+#'
+#' @param inSCE The single cell experiment to use.
+#' @param useAssay The assay to use.
+#' @param features A string or vector of strings with each gene to aggregate.
+#' @param displayName A string that is the name of the column used for genes.
+#' @param clusters The name of a colData entry that can be used as groups.
+#' @param title The title of the bubble plot
+#' @param xlab The x-axis label
+#' @param ylab The y-axis label
+#' @param colorLow The color to be used for lowest value of mean expression
+#' @param colorHigh The color to be used for highest value of mean expression
+#' @return A ggplot of the bubble plot.
+#' @importFrom rlang .data
+#' @examples
+#' data("scExample")
+#' plotBubble(inSCE=sce, useAssay="counts", features=c("B2M", "MALAT1"), displayName="feature_name", 
+#' clusters="type", title="cell type test", xlab="gene", ylab="cluster", 
+#' colorLow="white", colorHigh="blue")
+#' @export
+plotBubble <- function(inSCE, useAssay="logcounts", features, displayName=NULL, clusters="cluster", title="", xlab=NULL, ylab=NULL, colorLow="white", colorHigh="blue"){
+  metrics <- runClusterSummaryMetrics(inSCE, useAssay=useAssay, features=features, 
+                                      displayName=displayName, clusters=clusters)
+  .ggBubble(metrics$features, metrics$avgExpr, metrics$percExpr, colorLow, colorHigh, title)
+}
+
+.ggBubble <- function(features, avgExpr, percExpr, colorLow="white", colorHigh="blue", title=""){
```

Comment:
avgExpr and percExpr are data frames correct? So you could get features from rownames(avgExpr) and cluster names from colnames(avgExpr). We may just want to be consistent on how we allow users to specify these in .ggBubble. 

```
.ggBubble <- function(avgExpr, percExpr, features = NULL, groups = NULL, ...)
```

Then in the code have something like this:

```
if(is.null(features)) {
  if(is.null(rownames(avgExpr))) {
    stop("'features' must be supplied or the 'rownames' of the average expression matrix must be set.")
  }
  features <- rownames(avgExpr)
} else {
  ## Error checking for feature length
  if(length(features) != nrow(avgExpr)) {
    stop("'features'" must be the same length as the number of rows in the average expression matrix.")
  }
}
```

You can do something similar with groups (I think we should call them groups instead of clusters in this case).